research/src/pages/file.proposal.mdx

```diff
@@ -0,0 +1,178 @@
+
+### Appearance <a href="#appearance" id="appearance"></a>
+
+<Image alt="File Chooser Example" src="file.png" />
+
+### Anatomy <a href="#anatomy" id="anatomy"></a>
+
+TODO
+
+#### Diagram <a href="#diagram" id="diagram"></a>
+
+TODO
+
+#### DOM Structure <a href="#dom-structure" id="dom-structure"></a>
+
+```html
+<button tabindex="-1" part="file-selector-button">
+  <slot name="button">
+    <!-- Localized "Browse..." string -->
+  </slot>
+</button>
+<label part="file-selector-label">
+  <slot name="label">
+    <!-- implementation and state-dependent -->
+  </slot>
+</label>
+```
+
+The button is not tab-navigable, as the input is focused atomically, see below.
+
+#### CSS Parts <a href="#css-parts" id="css-parts"></a>
+
+| Part Name              | Description                                                                                         |
+| ---------------------- | --------------------------------------------------------------------------------------------------- |
+| `file-selector-button` | Enables styling the `&lt;button>` part without requiring it to be completely replaced via its slot. |
+| `file-selector-label`  | Enables styling the `&lt;label>` part without requiring it to be completely replaced via its slot.  |
+
+---
+
+## Behavior <a href="#behavior" id="behavior"></a>
+
+### States and Interactions <a href="#states-interactions" id="states-interactions"></a>
+
+- As a result of keyboard (either Enter or Spacebar without modifier) or mouse
+  / touch interaction (click), an implementation-dependent dialog will show
+  up. This dialog cannot be accessed by the website, but its contents and
+  appearance may depend on things like the `accept` attribute.
+
+- Dragging an file into the component may also be used to select a particular
+  file.
+
+## Accessibility <a href="#accessibility" id="accessibility"></a>
+
+File Inputs should be focusable via a keyboard navigation.
+
+#### Keyboard Navigation and Focus <a href="#keyboard-navigation" id="keyboard-navigation"></a>
+
+File input is focused atomically.
+
+<p class="question">
+
+Seems like ideally we could delegate focus to the button, or something something... Gecko used to
+do that, but I had to change it because the button exists as a result of layout (in Gecko, at
+least, that is). Also input type changes would be trickier with such a setup.
+See <a href="https://github.com/WICG/open-ui/issues/107">Issue #107</a>
+
+</p>
+
+#### Form Input <a href="#form-input" id="form-input"></a>
+
+This component integrates naturally with form submission as described in [the HTML spec](<https://html.spec.whatwg.org/#file-upload-state-(type=file)>).
+
+### Globalization <a href="#globalization" id="globalization"></a>
+
+The component reacts naturally to directionality. The initial label content and
+the button contents should be localized.
+
+### Security <a href="#security" id="security"></a>
+
+The component very intentionally doesn't expose to the outside world anything
```

Comment:
This should be written in a normative form. Possibly something like:

> The implementation should ensure that the component does not provide access to content on the user's file system unless the user has selected it.

Reply:
Also, there may be normative text in WHATWG that is similar to this and you may just want to link to that